feat: no-unknown-at-rules rule (#7)
eslint/css: docs/rules/no-unknown-at-rules.md

# no-unknown-at-rules

Disallow unknown at-rules.

## Background

CSS contains a number of at-rules, each beginning with a `@`, that perform various operations. Some common at-rules include:

-   `@import`
-   `@media`
-   `@font-face`
-   `@keyframes`
-   `@supports`
-   `@namespace`
-   `@page`
-   `@charset`

It's important to use a known at-rule because unknown at-rules cause the browser to ignore the entire block, including any rules contained within. For example:

```css
/* typo */
@charse "UTF-8";
```

> Here, the `@supports` at-rule is incorrectly spelled as `@support`, which means that the rule for `.grid-container` will be ignored even if the browser supports `display: grid`.

Reviewer: ```suggestion
Here, the `@charset` at-rule is incorrectly spelled as `@charse`, which means that it will be ignored.
```

# no-unknown-at-rules

Disallow unknown at-rules.

## Background

CSS contains a number of at-rules, each beginning with a `@`, that perform various operations. Some common at-rules include:

-   `@import`
-   `@media`
-   `@font-face`
-   `@keyframes`
-   `@supports`
-   `@namespace`
-   `@page`
-   `@charset`
-   `@document`

> -   `@viewport`

Reviewer: ```suggestion
```

`@viewport` is not in the list we get from `css-tree/definition-syntax-data` and I believe it's dropped from the spec.

It's important to use a known at-rule because unknown at-rules cause the browser to ignore the entire block, including any rules contained within. For example:

```css
/* typo */
@support (display: grid) {
	.grid-container {
		display: grid;
	}
}
```

Here, the `@supports` at-rule is incorrectly spelled as `@support`, which means that the rule for `.grid-container` will be ignored even if the browser supports `display: grid`.

## Rule Details

This rule warns when it finds a CSS at-rule that isn't part of the CSS specification. The at-rule data is provided via the [CSSTree](https://github.com/csstree/csstree) project.

Examples of incorrect code:

```css

> @support (display: grid) {

Reviewer: This is reported as a parsing error.
